### DISCUSSION TITLE
docs/g4install/chap2_guide/windows/g4install-111-win11/

### DISCUSSION THREAD

**Discussion by giscus** [evandde/blog_comment#63] [Announcements]
# docs/g4install/chap2_guide/windows/g4install-111-win11/

설치 상세: Windows 10/11, Geant4-11.1.0, Qt5 연동, Multi-threading 활성화

https://evanote.mew.kr/docs/g4install/chap2_guide/windows/g4install-111-win11/

**Comment by LE2E2** [comment]
박사님 안녕하세요. 한참 전에 올려주신 자료이지만 감사히 잘 보고 있습니다. 
알려주신 대로 따라하다가 마지막 빌드 과정에서(cmake ..) 아래와 같은 에러가 발생했습니다..
설치되고 대기하는 동안 자리를 비워 설치진행 창을 계속 확인하고 있지는 못했는데 그 와중에 문제가 발생한 것인지.. 혹시 한번 살펴봐 주시면 감사드리겠습니다. 

-- Building for: Visual Studio 17 2022
-- The C compiler identification is MSVC 19.39.33520.0
-- The CXX compiler identification is MSVC 19.39.33520.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2022/BuildTools/VC/Tools/MSVC/14.39.33519/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2022/BuildTools/VC/Tools/MSVC/14.39.33519/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:13 (find_package):
  By not providing "FindGeant4.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Geant4", but
  CMake did not find one.

  Could not find a package configuration file provided by "Geant4" with any
  of the following names:

    Geant4Config.cmake
    geant4-config.cmake

  Add the installation prefix of "Geant4" to CMAKE_PREFIX_PATH or set
  "Geant4_DIR" to a directory containing one of the above files.  If "Geant4"
  provides a separate development package or SDK, be sure it has been
  installed.

**Reply by evandde** [reply]
마지막 빌드과정이라는 부분이 "예제 실행해보기" 쪽인것 같습니만, 확실하지가 않네요.

현재 발생한 오류는 환경변수가 제대로 설정되지 않았을 경우 생기는 오류로 보입니다.

[환경변수 설정](https://evanote.mew.kr/doc_g4install/chap2_guide/windows/g4install-111-win11/#%ED%99%98%EA%B2%BD%EB%B3%80%EC%88%98-%EC%84%A4%EC%A0%95-1) 부분을 진행하신것이 맞는지 다시한번 되짚어보시기 바랍니다.

이 부분을 정상적으로 하였음에도 문제가 생긴다면, 관리자권한의 PowerShell에서 아래의 문구를 추가적으로 입력해보시기 바랍니다.

```bash
[Environment]::SetEnvironmentVariable("Geant4_DIR", "$env:USERPROFILE\Geant4\geant4-v11.1.0\install\lib\Geant4-11.1.0", "Machine")
```

**Reply by LE2E2** [reply]
써주신 코드 복붙 하는 과정에서 Geant4 버젼을 고치지 않고 복붙한 부분이 있어서 생긴 문제였던 것 같습니다.
수정 후 마지막 GUI창까지 띄우는 데 방금 성공했습니다. 
늦은시간까지 답변 감사합니다!

**Comment by LE2E2** [comment]
박사님 안녕하세요. 며칠 전에 질문드렸는데 또 문제가 생겨서 문의드립니다ㅠㅠ
지난 번 집 컴퓨터에서는 잘 설치했고, 회사 컴퓨터에 설치하려는데   cmake --build . --config Release --target install 단계에서 아래와 같은 문제가 발생했습니다.

... \Geant4\geant4-v11.2.1\source\graphics_reps\src\HepPolyhedron.cc(3084,1): warning C4819: 현재 코드 페이지(949)에서 표시할 수 없는 문자가 파일에 들어 있습니다. 데이터가 손실되지 않게 하려면 해당 파일을 유니코드 형식으로 저장하십시오. [... \Geant4\geant4-v11.2.1\build\source\G4graphics_reps.vcxproj]

특이사항은 사내 방화벽 문제로 인해 QT 다운로드가 위 방법으로는 안되어서, "QT 오프라인"으로 검색해서 5.12.12 버젼을 설치했습니다. (msvc 2017)
qmake 정상작동도 확인했구요.. 어떤문제일지 혹시 아실까요?
 
좋은 자료 항상 감사히 보고 있습니다!

**Reply by evandde** [reply]
일단 말씀주신 C4819 warning은 에러는 아닙니다.
Geant4가 기본적으로 리눅스/Mac 환경에서 개발되어, Windows에서 활용 시 인코딩 관련 이슈가 발생할 수 있습니다.
이러한 인코딩 관련 경고가 C4819입니다.

일반적으로 그 외의 문제가 없었다면, 저 warning이 뜨더라도 설치 자체는 문제없이 진행될 것입니다.

혹시 설치가 제대로 이루어지지 않는다면, "error"라고 뜨는 문구를 확인해보시기 바랍니다.

**Reply by LE2E2** [reply]
오.. 그러네요. 재부팅하니 위 경고문은 뜨는데 설치는 제대로 진행되었습니다!
추가로 여쭙고 싶은 것이있습니다 . 회사 방화벽으로 인해 .tar.gz 데이터 파일 12개는 따로 받고 아래 명령어를 사용했는데요. (-DGEANT4_INSTALL_DATA=OFF)
cmake .. -DCMAKE_INSTALL_PREFIX="..\Geant4\geant4-v11.2.1\install" -DGEANT4_BUILD_MULTITHREADED=ON -DGEANT4_BUILD_MSVC_MP=ON -DGEANT4_USE_QT=ON
 
별도로 받은 12개의 데이터 파일들을 설치 후 어떻게 해야 적용되는지 아실까요? 어느 폴더에 넣으면 된다든가.. 하는 방법이 있을까 해서 문의드립니다.

**Reply by LE2E2** [reply]
바로 위 질문에 대해서는 아래와 같은 글을 검색해서, 동일한 파일에 일단 압축 푼 후 해당 폴더에 넣었습니다.
... and unpack them under the directory: /home/esh/Downloads/geant4.10.07-install/share/Geant4-10.7.0/data 

그 후 또 쭉 잘 진행되다가 가장 마지막 exampleB1.exe 실행 시, 
검은 화면에 Available UI session types : [Qt, tcsh, csh] 떴다가 새창에서 그림이 안보이고 아래 빨간 글씨가 나타납니다

ERROR: Can not open a macro file <init_vis.mac>. Set macro path with "/control/macroPath" if needed.

-------- WWWW ------- G4Exception-START -------- WWWW -------
*** G4Exception : UIMAN0123
      issued by : G4UImanager::ApplyCommand
Command aborted (400)
Error code : 400
*** This is just a warning message. ***
-------- WWWW -------- G4Exception-END --------- WWWW -------

이와 같이 뜨면서 그림이 보이지 않습니다..ㅠㅠ 방법이 있을까요.

**Reply by evandde** [reply]
1. 데이터를 별도로 다운받아 연동하는 방법에 관해서는 왼쪽 설치가이드 중에 Windows 10/11 11.0.0 버전설치글을 참고하세요 ([링크](https://evanote.mew.kr/doc_g4install/chap2_guide/windows/g4install-110-win11/#cross-section-data-%EC%A4%80%EB%B9%84)). 데이터를 어느 경로에 두든 상관은 없으나, 설치 시에 `-DGEANT4_INSTALL_DATADIR="$env:USERPROFILE\Geant4\data"` 와 같은 명령어를 통해 데이터 경로를 잡아두면 됩니다. 혹은 검색을 통해 찾으신 바와 같이 share/Geant4(버전)/data에 넣으면 그 위치를 기본경로로 찾으므로 이렇게 해도 정상 동작합니다.

2. init_vis.mac에 관해서는, ./Release/exampleB1.exe 명령어로 실행하지 않고, Release 폴더에 들어가셔서 직접 실행하신것이 아닌가 추측됩니다.
exampleB1은 exampleB1.exe 프로그램을 실행하였을 때 그 동일경로에서 init_vis.mac 파일을 찾아오도록 코딩되어 있습니다.
아마, 제가 말씀드린걸 그대로 따라가셨다면, build 폴더 안에 init_vis.mac 파일이 있을 것입니다.
이 때문에, Release 폴더까지 들어가서 프로그램을 실행시키면 이런 문제가 발생합니다. (MSVC 컴파일러 특성으로 인해 윈도우에서만 겪는 번거로운 문제입니다)
해결책은 몇가지가 있습니다만, 제일 권장드리는 방법은 실행을 build 폴더 내에서 `./Release/exampleB1.exe` 명령어로 실행시키는 것입니다.

**Reply by LE2E2** [reply]
흐어.. 맞습니다. 더블클릭으로 실행시켜서 안되는 거였군요. 말씀하신 대로 명령어로 하니 해결되었습니다..! 
퇴근할 수 있겠습니다 ㅎㅎ 감사합니다. 행복한 주말 보내세요!!
